### PR TITLE
[release-3.4] Move go version to dedicated .go-version file

### PR DESCRIPTION
**Pull request by serathius**
Backport of https://github.com/etcd-io/etcd/pull/15859 
cc @ahrtr @ptabor

### CODE REVIEW

File: .go-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.19.9
```
Comment (fuweid) [inline .go-version:1]: It seems like there is unexpected `No newline at end of file`.

**Review by fuweid — APPROVED** [review]
LGTM

**Review by jmhbnz — APPROVED** [review]
LGTM

**Review by ahrtr — APPROVED** [review]
LGTM

**Pull request merged**